File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/telegram/ChosenInlineResultFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.telegram;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.StringFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface ChosenInlineResultFilter {

    UserFilter from() default @UserFilter(AnnotationState.OFF);

    LocationFilter location() default @LocationFilter(AnnotationState.OFF);

    StringFilter inlineMessageId() default @StringFilter(AnnotationState.OFF);

    StringFilter query() default @StringFilter(AnnotationState.OFF);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/composite/ChosenInlineResultFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.composite

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class ChosenInlineResultFilter(
        val from: UserFilter = UserFilter(status = AnnotationState.OFF),
        val validator: Array<String> = [],
        val location: LocationFilter = LocationFilter(status = AnnotationState.OFF),
        val inlineMessageId: StringFilter = StringFilter(status = AnnotationState.OFF),
        val query: StringFilter = StringFilter(status = AnnotationState.OFF)
)
